src/materialized/bin/materialized.rs

```diff
@@ -288,3 +291,73 @@ to improve both our software and your queries! Please reach out at:
 "
     );
 }
+
+/// Attempts to increase the soft nofile rlimit to the maximum possible value.
+fn adjust_rlimits() {
+    // getrlimit/setrlimit can have surprisingly different behavior across
+    // platforms, even with the rlimit wrapper crate that we use. This function
+    // is chattier than normal at the trace log level in an attempt to ease
+    // debugging of such differences.
+
+    let (soft, hard) = match rlimit::Resource::NOFILE.get() {
+        Ok(limits) => limits,
+        Err(e) => {
+            trace!("unable to read initial nofile rlimit: {}", e);
+            return;
+        }
+    };
+    trace!("initial nofile rlimit: ({}, {})", soft, hard);
+
+    #[cfg(target_os = "macos")]
+    let hard = {
```

Comment:
TIOLI: I slightly prefer ifdefs only at the function level, so having something like:

```rust
fn adjust_rlimits() {
    ..
    let hard = get_desired_hardlimit(hard);
    ..
}

#[cfg(target_os = "macos")]
fn get_desired_hardlimit(lim: usize) -> Result<usize, _> { ..}

#[cfg(not(target_os = "macos"))]
fn get_desired_hardlimit(lim: usize) -> Result<usize, _> { lim }
```

Reply:
Heh, I think we just have different preferences for something like this. Going to leave as is.